sequence/sequence_generator.hpp

```diff
@@ -235,10 +235,10 @@ class SeqGen {
             }
             case 2: { // substitute
                 auto pos = rpos_exc(gen);
-                auto c = rchar(gen);
+                auto c = ins_char(gen);
                 if (c == ref[pos]) {
                     c++;
-                    c = (c % seq_len);
+                    c = (c % alphabet_size);
```

Comment:
I don't think this is right, as the chances that the substituted character is `ref[pos]+1` are double, so the mutation is not uniform. Take a look at my implementation of `point_mutate` (line 201) for what I think is a true uniform distribution.

Reply:
yes you're right, in the line just above if (auto c = ... ) must be "c=sub_char(gen)" instead of "c=ins_char(gen)", this will pick the character from set {0,...,alphabet_size-2}, which would resolve the problem.